### DISCUSSION TITLE
Once installed a firebase link always returns a link

### DISCUSSION THREAD

**Discussion by uguraktas** [invertase/react-native-firebase#6278] [General]
I am using firebase deeplink with react navigation. There is no problem in the operation of the redirect and deeplink. But if I installed the app with a deeplink, that deeplink comes every time I close and open the app even when I change nav stack, I am directed to that screen again.

My code same with this:
https://reactnavigation.org/docs/deep-linking#third-party-integrations 

I did some research. There is something called `DynamicLink.matchType` but I couldn't find it in react-native-firebase. How exactly can I fix this issue?
https://stackoverflow.com/a/47998881/8519853

```
async getInitialURL() {
            const { isAvailable } = utils().playServicesAvailability

            if (isAvailable) {
                const initialLink = await dynamicLinks().getInitialLink(); // <--- this place return url every time I close and open the app

                if (initialLink) {
                    return initialLink.url;
                }
            }


            const url = await Linking.getInitialURL();
        
            return url;
        },
```

**Comment by uguraktas** [comment]
Is there any solution about it?

**Reply by mikehardy** [reply]
Hi there - this did not really contain enough details to say
https://stackoverflow.com/help/mcve / https://stackoverflow.com/help/how-to-ask

match type is only valid on ios, not android, so it was not cross-platform and was not implemented. I don't see the behavior you describe on either android or ios but I have not tested the app-install-via-deep-link case in a really long time.

If you isolate it to one platform or the other, this seems like the sort of thing you should take up with the underlying SDK (either firebase-android-sdk or firebase-ios-sdk - both have repos on github) after verifying it reproduces with one of their quickstarts (either https://github.com/mikehardy/quickstart-android/blob/master/dynamiclinks/README.md or https://github.com/firebase/quickstart-ios/blob/master/dynamiclinks/README.md )
